src/Syntax_Server/Oni_Syntax_Server.re

```diff
@@ -151,38 +153,46 @@ let start = (~healthCheck) => {
           | Message(protocol) => handleProtocol(protocol);
 
         while (isRunning^) {
-          log("Waiting for incoming message...");
-
-          // Wait for pending incoming messages
-          Mutex.lock(messageMutex);
-          while (!hasPendingMessage() && !State.anyPendingWork(state^)) {
-            Condition.wait(messageCondition, messageMutex);
-          };
-
-          // Once we have them, let's track and run them
-          let messages = flush();
-          Mutex.unlock(messageMutex);
-
-          // Handle messages
-          messages
-          // Messages are queued in inverse order, so we need to fix that...
-          |> List.rev
-          |> List.iter(handleMessage);
-
-          // TODO: Do this in a loop
-          // If the messages incurred work, do it!
-          if (State.anyPendingWork(state^)) {
-            log("Running unit of work...");
-            map(State.doPendingWork);
-            log("Unit of work completed.");
-          } else {
-            log("No pending work.");
+          try(
+            {
+              log("Waiting for incoming message...");
```

Comment:
I don't know if this is part of your plan already, but I'd really like to see this `log` function disappear in favor of using `Timber` with a custom reporter that forwards messages to the main process, as that would give access to all its logging features. Most importantly the severity, but we could also set the log level so we can increase the granularity if needed without affecting performance.

Reply:
Good call! Yes, I plan on using `Timber` and adding a file-logging option. 

I'm considering removing the log forwarding, it's a bit misleading given that the timing is interleaved incorrectly - I think it'd actually be more understandable as a separate file. The important parts of the interaction for most cases can be gleaned from the editor's log - the guts of the syntax server are most important for debugging, IMO. I'll work on a PR and we can discuss more on that.